### PR TITLE
feature: add network compress && set max value of read message size 2…

### PR DESCRIPTION
**Pull request by LucasLi1024**
compression ratio to be tested while opening SSL and testing it by sdk-demo PerformanceDMC

### CODE REVIEW

File: bcos-boostssl/websocket/WsConfig.h [hunk 1]
```diff
@@ -90,6 +90,8 @@ class WsConfig
     // cert config for boostssl
     std::shared_ptr<context::ContextConfig> m_contextConfig;
 
+    uint64_t m_maxReadMsgSize;
```
Comment (cyjseagull) [inline bcos-boostssl/websocket/WsConfig.h]: set a default value

File: bcos-boostssl/websocket/WsService.cpp [hunk 2]
```diff
@@ -296,6 +296,7 @@ bool WsService::registerMsgHandler(uint32_t _msgType, MsgHandler _msgHandler)
 
 std::shared_ptr<WsSession> WsService::newSession(std::shared_ptr<WsStream> _stream)
 {
+    _stream->setReadMaxMsg(m_config->maxReadMsgSize());
```
Comment (cyjseagull) [inline bcos-boostssl/websocket/WsService.cpp:299]: Pls make sure the option effective here

File: bcos-boostssl/websocket/WsConnector.cpp [hunk 3]
```diff
@@ -238,6 +244,12 @@ void WsConnector::connectToWsServer(const std::string& _host, uint16_t _port,
             auto stream = std::make_shared<boost::beast::websocket::stream<
                 boost::beast::ssl_stream<boost::beast::tcp_stream>>>(*ioc, *ctx);
 
+            // must set compress option before handshake, otherwise network compress will fail
+            boost::beast::websocket::permessage_deflate opt;
```
Comment (cyjseagull) [inline bcos-boostssl/websocket/WsConnector.cpp]: you can encapsulated this as a function.

File: bcos-boostssl/websocket/WsService.cpp (hunk 2 above)
Comment (LucasLi1024) [reply bcos-boostssl/websocket/WsService.cpp:299]: the option had been proved be effective by testing message size using sample in test directory.

File: bcos-boostssl/websocket/WsConfig.h (hunk 1 above)
Comment (LucasLi1024) [reply bcos-boostssl/websocket/WsConfig.h]: has set a default value as 16M

File: bcos-boostssl/websocket/WsConnector.cpp (hunk 3 above)
Comment (LucasLi1024) [reply bcos-boostssl/websocket/WsConnector.cpp]: has encapsulated a function in Wstools.h

**Review by cyjseagull — APPROVED** [review]

**Pull request merged**